lib/shrine.rb

```diff
@@ -799,9 +799,14 @@ def download(*args)
           if storage.respond_to?(:download)
             storage.download(id, *args)
           else
-            tempfile = Tempfile.new(["shrine", ".#{extension}"], binmode: true)
-            open(*args) { |io| IO.copy_stream(io, tempfile.path) }
-            tempfile.tap(&:open)
+            begin
+              tempfile = Tempfile.new(["shrine", ".#{extension}"], binmode: true)
+              open(*args) { |io| IO.copy_stream(io, tempfile.path) }
+              tempfile.tap(&:open)
+            rescue => err
+              tempfile.close!
+              raise err
```

Comment:
If we call `raise` without an argument, Ruby will automatically raise the last rescued exception, so we could omit the exception variable here:

```rb
begin
  # ...
rescue
  # ...
  raise
end
```

Reply:
I did not know this and actually find it ambiguous. I like Ruby's tolerance for omitting things like parenthesis and stuff to reduce visual clutter in the code but I tend to draw the line when it makes the code ambiguous. If I saw this I'd think the code is raising raise's default exception and not the rescued exception. I'd have to consult the docs and/or know that Ruby is actually raising the rescued error.

That's my opinion however I'll go ahead and change it as that's subjective. If you change your mind, I can change it back too - not a problem.